test/sequential/test-uv-threadpool-schedule.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+
+// test to validate massive dns lookups do not block filesytem I/O
+// (or any fast I/O).Prior to https://github.com/libuv/libuv/pull/1845
+// few back-to-back dns lookups were sufficient to engage libuv
+// threadpool workers in a blocking manner, throttling other work items
+// that need pool resources. Start slow and fast I/Os together, make sure
+// fast I/O can complete in at least in 1/100th of time for slow I/O.
+// TEST TIME TO COMPLETION: ~5 seconds.
+
+const common = require('../common');
+const dns = require('dns');
+const fs = require('fs');
+const assert = require('assert');
+
+const start = Date.now();
+
+const slowIOmax = 100;
+let slowIOcount = 0;
+let fastIOdone = false;
+let slowIOend, fastIOend;
+
+for (let i = 0; i < slowIOmax; i++) {
+  // We need to refresh the domain string everytime,
+  // otherwise the TCP stack that cache the previous lookup
+  // returns result from memory, breaking all our Math.
```

Comment:
In that case, using a base string and then keep incrementing an offset would guarantee that there will be no repetition in the domain names used, right? Because, the random numbers might repeat (even though odds of that happening is less)

Reply:
thanks for the suggestion, followed your suggestion.

Reply:
@thefourtheye - I just spotted a drawback of doing this: if you run the test back-to-back, it fails the second time onwards - for obvious reasons (the domain strings being static, they get resolved faster). I don't know if it affects our CI tests though.